sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py

```diff
@@ -310,43 +312,50 @@ async def list(
 
-    @distributed_trace_async
-    async def enable(self, **kwargs) -> bool:
-        """Enable Application Insights tracing.
-        This method makes service calls to get the properties of the Applications Insights resource
-        connected to the Azure AI Studio Project. If Application Insights was not enabled for this project,
-        this method will return False. Otherwise, it will return True. In this case the Application Insights
-        connection string can be accessed via the `.diagnostics.connection_string` property.
-
-        :return: True if Application Insights tracing was enabled. False otherwise.
-        :rtype: bool
+    async def get_connection_string(self) -> str:
         """
+        Get the Application Insights connection string associated with the Project's Application Insights resource.
+        On first call, this method makes a GET call to the Application Insights resource URL to get the connection string.
+        Subsequent calls return the cached connection string.
 
-        if not self.connection_string:
-            # Get the AI Studio Project properties
+        :return: The connection string, or `None` if an Application Insights resource was not enabled for the Project.
+        :rtype: str
+        """
+        if not self._get_connection_string_called:
+            # Get the AI Studio Project properties, including Application Insights resource URL if exists
             get_workspace_response: GetWorkspaceResponse = await self._outer_instance.connections._get_workspace()
 
-            # No Application Insights resource was enabled for this Project
-            if not get_workspace_response.properties.application_insights:
-                return False
+            # Continue only if Application Insights resource was enabled for this Project
+            if get_workspace_response.properties.application_insights:
 
-            app_insights_respose: GetAppInsightsResponse = await self.get_app_insights(
-                app_insights_resource_url=get_workspace_response.properties.application_insights
-            )
+                # Make a GET call to the Application Insights resource URL to get the connection string
+                app_insights_respose: GetAppInsightsResponse = await self.get_app_insights(
+                    app_insights_resource_url=get_workspace_response.properties.application_insights
+                )
 
-            if not app_insights_respose.properties.connection_string:
-                raise ValueError("Application Insights resource does not have a connection string")
+                self._connection_string = app_insights_respose.properties.connection_string
 
-            self.connection_string = app_insights_respose.properties.connection_string
+        self._get_connection_string_called = True
+        return self._connection_string
 
-        return True
+
+    # TODO: what about `set AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED=true`?
```

Comment:
Checking if these TODO comments are still valid?

Reply:
Yes, thanks, still valid. I need to discuss this further with Dan, Johan, Liudmila and Trupti.